packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -869,7 +884,18 @@ export function transformCjsImport(
     }) as any
   ).body[0] as Node
 
+  // `export * from '...'` may cause unexpected problem, so give it a warning
   if (
+    config.command === 'serve' &&
+    node.type === 'ExportAllDeclaration' &&
+    !node.exported
+  ) {
+    config.logger.warn(
+      colors.yellow(
+        `\n\`${importExp}\` may lose module exports after pre-bundling origin cjs module "${rawUrl}", please use named exports(e.g \`export {A,B} from "${rawUrl}"\`) instead.`,
```

Comment:
This fix makes sense to me. I thought the prebundle metadata has information of the exports, but it makes sense it doesn't because CJS is too dynamic.

I think we can also put a note here that the dependency can export as ESM to workaround the issue too.

```suggestion
        `Unable to interop \`${importExp}\` and may lose module exports. Please export "${rawUrl}" as ESM or use named exports instead, \`export { A, B } from "${rawUrl}"\``,
```

I think it would be nice to tell what file causes this too. Even better if there's a code trace. But if that's a bit tricky to implement I'm fine with this message as well for now.

Reply:
I added the `importer` info to the warning.